model/jpa/src/main/java/org/keycloak/connections/jpa/PersistenceExceptionConverter.java

```diff
@@ -27,31 +29,52 @@
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.Proxy;
+import java.util.regex.Pattern;
 
 /**
  * @author <a href="mailto:sthorger@redhat.com">Stian Thorgersen</a>
  */
 public class PersistenceExceptionConverter implements InvocationHandler {
 
-    private EntityManager em;
+    private static final Pattern WRITE_METHOD_NAMES = Pattern.compile("persist|merge");
 
-    public static EntityManager create(EntityManager em) {
-        return (EntityManager) Proxy.newProxyInstance(EntityManager.class.getClassLoader(), new Class[]{EntityManager.class}, new PersistenceExceptionConverter(em));
+    private final EntityManager em;
+    private final boolean batchEnabled;
+    private final int batchSize;
+    private int changeCount = 0;
+
+    public static EntityManager create(KeycloakSession session, EntityManager em) {
+        return (EntityManager) Proxy.newProxyInstance(EntityManager.class.getClassLoader(), new Class[]{EntityManager.class}, new PersistenceExceptionConverter(session, em));
     }
 
-    private PersistenceExceptionConverter(EntityManager em) {
+    private PersistenceExceptionConverter(KeycloakSession session, EntityManager em) {
+        batchEnabled = session.getAttributeOrDefault(Constants.STORAGE_BATCH_ENABLED, false);
+        batchSize = session.getAttributeOrDefault(Constants.STORAGE_BATCH_SIZE, 100);
```

Comment:
Do we have some performance results with different `STORAGE_BATCH_SIZE` values? I think we should set the default value based on best results, wdyt?

Reply:
Sure. That number is what worked well for the use case we are testing using 1k realms.

Personally, I don't think it is critical for this PR. We can always change the value and users are free to set any value they want.

There is more work in the backlog that could help with this. But they are a follow-up to this PR and the other one related to multi-realm.